cryptoprotocol/src/main/java/org/cryptimeleon/incentivesystem/cryptoprotocol/benchmark/Benchmark.java

```diff
@@ -0,0 +1,199 @@
+package org.cryptimeleon.incentivesystem.cryptoprotocol.benchmark;
+
+
+import org.cryptimeleon.craco.sig.sps.eq.SPSEQSignature;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.IncentiveSystem;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.*;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderKeyPair;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderPublicKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderSecretKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserKeyPair;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserPublicKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserSecretKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.messages.JoinRequest;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.messages.JoinResponse;
+
+import java.math.BigInteger;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.BiConsumer;
+
+/**
+ * Contains benchmark code.
+ */
+public class Benchmark {
+
+    /**
+     * Increase/decrease used for earn/spend in benchmark
+     */
+    private static final BigInteger EARN_SPEND_AMOUNT = BigInteger.TEN;
+
+    /**
+     * Run a benchmark with the given benchmarkConfig
+     *
+     * @param benchmarkConfig a configuration object that defines parameters of the benchmark
+     * @return a BenchmarkResult object that contains all data collected throught the benchmark
+     */
+    public static BenchmarkResult runBenchmark(
+            BenchmarkConfig benchmarkConfig
```

Comment:
Looks a little odd to me, why not having the method head in a single line? I see that the way you did it here is useful for long and complicated signatures but it looks weird for such a short one.

Reply:
Good point :)